paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -166,14 +166,43 @@ public Identifier identifier() {
 
     @Override
     public Optional<Statistics> statistics() {
-        // todo: support time travel
         Snapshot latestSnapshot = snapshotManager().latestSnapshot();
         if (latestSnapshot != null) {
             return store().newStatsFileHandler().readStats(latestSnapshot);
         }
         return Optional.empty();
     }
 
+    @Override
+    public Optional<Statistics> statistics(Long snapshotId) {
+        if (!snapshotManager().snapshotExists(snapshotId)) {
```

Comment:
Just `return store().newStatsFileHandler().readStats(latestSnapshot);`?

Reply:
Yes, it should return store().newStatsFileHandler().readStats(latestSnapshot) better. Thanks~ @JingsongLi